editor/plugins/tiles/tiles_editor_plugin.cpp

```diff
@@ -172,10 +175,13 @@ void TilesEditorPlugin::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_INTERNAL_PROCESS: {
 			if (tile_map_changed_needs_update) {
-				TileMap *tile_map = Object::cast_to<TileMap>(ObjectDB::get_instance(tile_map_id));
-				if (tile_map) {
-					tile_set = tile_map->get_tileset();
+				if (is_visible) {
```

Comment:
I don't feel this is right. What is this check for exactly?
If the TileMap changed, and the TileSet was changed for another one, I don't get why the visibility of the editor would have an impact on that. Is this for optimization?